packages/payment-processor/src/payment/erc777-utils.ts

```diff
@@ -253,3 +259,32 @@ export async function unwrapSuperToken(
   );
   return signer.sendTransaction(preparedTx);
 }
+
+/**
+ * Check if there's an existing stream for the trio (Sender, Receiver, Currency)
+ * @param sender address
+ * @param receiver address
+ * @param currency to stream
+ * @param provider web3 provider.
+ * @returns
+ */
+export async function checkExistingStream(
+  request: ClientTypes.IRequestData,
+  sender: string,
+  provider: providers.Provider = getProvider(),
+): Promise<boolean> {
+  const id = getPaymentNetworkExtension(request)?.id;
+  if (id !== ExtensionTypes.PAYMENT_NETWORK_ID.ERC777_STREAM) {
+    throw new Error('Not a supported ERC777 payment network request');
+  }
+  validateRequest(request, ExtensionTypes.PAYMENT_NETWORK_ID.ERC777_STREAM);
+  const { paymentAddress } = getRequestPaymentValues(request);
+  const sf = await getSuperFluidFramework(request, provider);
+  const streams = await sf.query.listStreams({
+    sender,
+    receiver: paymentAddress,
+    token: request.currencyInfo.value,
+  });
+
+  return streams.data.length === 1 && streams.data[0].currentFlowRate !== '0';
```

Comment:
just to challenge, why limited to 1? should it be `length > 0`?

Reply:
It's a limitation of the Superfluid protocol that it can only have one stream for the combination, (sender, receiver, token), but @leoslr should also confirm it's the case with this situation.